compiler/src/dotty/tools/dotc/semanticdb/TypeOps.scala

```diff
@@ -415,12 +415,13 @@ object SymbolScopeOps:
   import Scala3.{_, given}
   extension (syms: List[SemanticSymbol])
     def sscope(using linkMode: LinkMode)(using SemanticSymbolBuilder, TypeOps, Context): s.Scope =
-      linkMode match {
-        case LinkMode.SymlinkChildren =>
-          s.Scope(symlinks = syms.map(_.symbolName))
-        case LinkMode.HardlinkChildren =>
-          s.Scope(hardlinks = syms.map(_.symbolInfo(Set.empty)))
-      }
+      // if syms contains FakeSymbol, hardlink those symbols
+      // because fake symbols don't appear in Symbols section.
+      // If we symlink those fake symbols, we always fail to lookup those symlinked symbols.
+      if syms.exists(s => s.isInstanceOf[FakeSymbol]) || linkMode == LinkMode.HardlinkChildren then
+        s.Scope(hardlinks = syms.map(_.symbolInfo(Set.empty)(using LinkMode.HardlinkChildren)))
```

Comment:
I think instead of Hard Linking, they should just be registered in the symbols section (unless they are within a refinement)

Reply:
Registered into symbols section :tada: https://github.com/lampepfl/dotty/pull/12885/commits/8e7e764685a8485f970cf01792d43e26ace782b4